src/vos/tests/vts_aggregate.c

```diff
@@ -631,7 +631,7 @@ discard_4(void **state)
 {
 	struct io_test_args	*arg = *state;
 	struct agg_tst_dataset	 ds = { 0 };
-	daos_epoch_t		 punch_epoch[2];
+	daos_epoch_t		 punch_epoch[3] = {};
```

Comment:
I added a terminal 0 entry.  Otherwise, whatever garbage is on the stack might match in the checking function

Reply:
I don't get it, what checking function will check punch_epoch[2]? I suppose only punch_eopch[0] and punch_epoch[1] will be used/checked.

Reply:
I see, thanks. Then I think we'd just fix defect like "if (punch_nr > 0 && punch_idx < punch_nr && punch_epoch[punch_idx] == epoch)", what do you think?

Reply:
done